lib/api/consumers/format-to-legacy-helpers.js

```diff
@@ -332,3 +324,70 @@ const getAddressLegacyId = (addressLegacyIDSet, legacyInteropKey, suffix = 0) =>
 
   return `${legacyInteropKey}`
 }
+
+const getAddressLegacyInteropKey = (meta, {legacyCommonToponymId, number, suffix}) => {
+  const cleInterop = meta.bal?.cleInterop
+  const isValidInteropKey = checkIfCleInteropIsValid(cleInterop)
+  const addressLegacyInteropKey = isValidInteropKey
+    ? cleInterop
+    : `${legacyCommonToponymId}_${String(number).padStart(5, '0')}${suffix ? `_${suffix}` : ''}`.toLowerCase()
+
+  return addressLegacyInteropKey
+}
+
+const getLegacyCommonToponymId = (
+  commonToponymLegacyIDSet,
+  legacyCommonToponymFantoirId,
+  meta,
+  {cog, id, pseudoCodeVoieGenerator, legacyLabelValue, codeAncienneCommune}
+) => {
+  let legacyCommonToponymId
+  const cleInterop = meta.bal?.cleInterop
+  const isCleInteropValid = checkIfCleInteropIsValid(cleInterop)
+  if (isCleInteropValid) {
+    const lastUnderscoreIndex = cleInterop.lastIndexOf('_')
+    legacyCommonToponymId = cleInterop.slice(0, lastUnderscoreIndex)
```

Comment:
```suggestion
    const secondUnderscoreIndex = cleInterop.indexOf('_',6) //5 insee code + _
    legacyCommonToponymId = cleInterop.slice(0, secondUnderscoreIndex )
```
cf le checkIfCleInteropIsValid pour la composition mais pas forcement l'avant dernier